pkg/auth/github/provider.go

```diff
@@ -71,6 +80,16 @@ func (p *Provider) GetUserDetails(code string, user *auth.User) error {
 		}
 	}
 
+	if p.Teams != "" {
+		if p.Organization == "" {
+			return fmt.Errorf("organization is required for teams")
+		}
```

Comment:
This can be evaluated during the [config validation](https://github.com/terralist/terralist/blob/master/pkg/auth/github/config.go#L17), there is no need to perform this check on every request.

Also, if the config is invalid, the error will be shown when the server is launched instead of when this case happens.